std/assembly/array.ts

```diff
@@ -153,7 +153,20 @@ export class Array<T> extends ArrayBufferView {
   }
 
   includes(value: T, fromIndex: i32 = 0): bool {
-    return this.indexOf(value, fromIndex) >= 0;
+    if (isFloat<T>()) {
+      let length = this.length_;
+      if (length == 0 || fromIndex >= length) return false;
+      if (fromIndex < 0) fromIndex = max(length + fromIndex, 0);
+      let dataStart = this.dataStart;
+      while (fromIndex < length) {
+        let elem = load<T>(dataStart + (<usize>fromIndex << alignof<T>()));
+        if (elem == value || bool(i32(isNaN(elem)) & i32(isNaN(value)))) return true;
```

Comment:
I believe this can be written as just `if (a == b || isNaN(a) & isNaN(b)) return true;` since `&` on bools stays bool

Reply:
Ah, right, unfortunate